scripts/quests/ahtUrhgan/Two_Horn_the_Savage.lua

```diff
@@ -0,0 +1,138 @@
+-----------------------------------
+-- Two Horn the Savage
+-----------------------------------
+-- Log ID: 6, Quest ID: 17
+-----------------------------------
+-- !zone 50       = Whitegate
+-- Milazahn       = !pos -58.1921 0.000 1.9746
+-- Cacaroon       = !pos -71.1544 0.000 -86.0422
+-- !zone 65       = Mamook
+-- Viscous Liquid = !pos -262.437 5.130 -141.241
+-----------------------------------
+local ID = zones[xi.zone.MAMOOK]
+-----------------------------------
+
+local quest = Quest:new(xi.questLog.AHT_URHGAN, xi.quest.id.ahtUrhgan.TWO_HORN_THE_SAVAGE)
+
+quest.reward =
+{
+    item = xi.item.IMPERIAL_MYTHRIL_PIECE,
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_AVAILABLE
+        end,
+
+        [xi.zone.AHT_URHGAN_WHITEGATE] =
+        {
+            ['Milazahn'] = quest:progressEvent(594),
+
+            onEventFinish =
+            {
+                [594] = function(player, csid, option, npc)
+                    if option == 1 then
+                        quest:begin(player)
+                    end
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_ACCEPTED
+        end,
+
+        [xi.zone.AHT_URHGAN_WHITEGATE] =
+        {
+            ['Cacaroon'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Temp') == 1 then
+                        return quest:event(608):oncePerZone()
+                    end
+                end,
+
+                onTrade = function(player, npc, trade)
+                    if
+                        quest:getVar(player, 'Prog') == 0 and
+                        npcUtil.tradeHasExactly(trade, { { 'gil', 1000 } })
+                    then
+                        return quest:progressEvent(595)
+                    end
+                end,
+            },
+
+            ['Milazahn'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 4 then
+                        return quest:progressEvent(596)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [596] = function(player, csid, option, npc)
+                    quest:complete(player)
+                end,
+
+                [595] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Temp', 1)
+                    quest:setVar(player, 'Prog', 1)
+                end,
+            },
+        },
+
+        [xi.zone.MAMOOK] =
+        {
+            ['Viscous_Liquid'] =
+            {
+                onTrigger = function(player, npc)
+                    local progress = quest:getVar(player, 'Prog')
+                    local liquid = npc:getID()
```

Comment:
no need for the ocal here, its only being called 1x, just move the logic to the check
```lua
if npc:getID() == ID.npc.QUEST_LIQUID then
```